src/StableHashTraits.jl

```diff
@@ -61,65 +152,58 @@ See also: [`StableHashTraits.hash_method`](@ref).
 """
 write(io, x, context) = write(io, x)
 write(io, x) = Base.write(io, x)
-function stable_hash_helper(x, hash, context, ::UseWrite)
+function stable_hash_helper(x, hash_state, context, ::WriteHash)
     io = IOBuffer()
     write(io, x, context)
-    update!(hash, take!(io))
-    return hash
+    update!(hash_state, take!(io))
+    return hash_state
 end
 
-function recursive_hash!(hash, result)
-    interior_hash = digest!(result)
-    # digest will return nothing if no objects have been added to the hash when using
-    # GenericFunHash; in this case, don't update the hash at all
-    if !isnothing(interior_hash)
-        update!(hash, copy(reinterpret(UInt8, vcat(interior_hash))))
-    end
-    return hash
+function recursive_hash!(hash_state, nested_hash_state)
+    nested_hash = digest!(nested_hash_state)
+    update!(hash_state, copy(reinterpret(UInt8, vcat(nested_hash))))
```

Comment:
This was required for CRC32c. In reality I don't think we ever really want to use it, and it can still work if you pass `alg=(x, s=0x000000) -> CRC32c.crc32c(copy(x), s)`, so I will remove the copy.